peps/pep-0772.rst

```diff
@@ -0,0 +1,1612 @@
+python`` are unlikely to be of benefit since the environment tends to be less
+reliable than our search.
+
+If no tag is yet requested, the ``VIRTUAL_ENV`` environment variable will be
+consulted to see if an environment has been activated. If so, that will become
+the request.
+
+If a tag has been requested at this stage, the ``python3`` command will verify
+that it matches ``PythonCore\\3.*`` and exit with an error if not. This allows
+allows the ``python3`` command to be used in an active environment consistent
+with other platforms, but not if the environment would not have included the
+command. This applies to most existing versions of Python on Windows. (The
+alternative to this behaviour is to make ``python3`` always error when an
+environment is active, as anything else would behave inconsistently for the
+user.)
+
+If no tag is requested, the default will be consulted. For ``python3``, this is
+``PythonCore\\3``, but for all other commands it is read from configuration
+(which might involve an environment variable). If it's still empty, any tag will
+be allowed.
+
+The best installed runtime matching the tag is then selected and launched with
+the remaining command line.
+
+If no matching runtime is found, and the ``--install`` option is set, it will be
+installed and then launched. This is the default for ``py exec``, and also for
+``python`` but only when no runtime was requested (which implies that no
+runtimes were installed). In all other cases, if no install is found, an error
+is printed and the process exits.
+
+
+Install subcommand
+------------------
+
+.. code:: text
+
+   py install [-s|--source <URL>] [-f|--force] [-u|--upgrade] [tag ...]
+   py install [-s|--source <URL>] [-t|--target <DIR>] [tag ...]
+   py install [-s|--source <URL>] [-d|--download <DIR>] [tag ...]
+   py install --refresh
+
+.. note::
+   This and all later subcommands are also available under ``pymanager``.
+   However, as we intend for ``py`` to be the usual command, we only show that
+   one.
+
+This subcommand will install one or more runtimes onto the current machine.
+The tags are ``Company\\Tag`` pairs (or just ``Tag`` if no slash is included),
+and are used to search the index file. Company names match as case-insensitive
+prefixes, preferring a full match over a prefix, and tags use case-insensitive,
+number-aware matches, with dotted numbers treated as versions. Tags must match
+one of the listed "install for" tags, and entries list multiple such tags to
+handle abbreviated requests.
+
+For example, the ``3.10.5`` entry would list all of ``3``, ``3.10`` and
+``3.10.5`` as tags to be installed for. A request for ``3.10`` would match one
+of these and so the entry is selected. Due to the number-aware matches, a
+request for ``03.0010`` would also match, and ``3.10.50`` would not.
+
+Tags may also be specified as a constraint, using ``>``, ``>=``, ``<``, ``<=``
+or ``!=`` followed by the ``Company\\Tag`` or ``Tag`` value. When matching a
```

Comment:
Is `~=` allowed?

Reply:
Great question (for the discussion thread ;) ) (but not yet, and not planned, unless someone suggests/motivates it)